### PR TITLE
Fix unable to load shader source files

### PR DESCRIPTION
**Pull request by gruberchris**
A few fixes for bugs I found while using this template with Visual Studio and CMake.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
+CMakeLists.txt.user
```
Comment (ArthurSonzogni) [inline .gitignore]: I consider `.gitignore` to be an antipattern. I never use them.
I think developers should control what they add and not use `git add .`

Could you please undo this change?

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required (VERSION 3.1)
 project (opengl-skeleton)
 
 # The main executable
-add_executable(main
+add_executable(${CMAKE_PROJECT_NAME}
```
Comment (ArthurSonzogni) [inline CMakeLists.txt]: Could you undo this?

(× multiple time)

File: src/Shader.cpp [hunk 3]
```diff
@@ -21,7 +21,7 @@ using namespace glm;
 // file reading
 void getFileContents(const char* filename, vector<char>& buffer) {
   ifstream file(filename, ios_base::binary);
-  if (file) {
+  if (file.good()) {
```
Comment (ArthurSonzogni) [inline src/Shader.cpp]: What does this bring?

**Review by ArthurSonzogni — COMMENTED** [review]
Thanks!
This is looking good. I just have a few comments below:

File: src/Shader.cpp (hunk 3 above)
Comment (gruberchris) [reply src/Shader.cpp]: I may have confused myself here about evaluating **file**. I had thought this **ifstream** variable was never going to be null and I think that is true that you can't check if its null (I've done a lot more Nodejs than C++ in recent years) but I think what you had was working because after c++ 11 that evaluates to true or false, doesn't it? 

https://stackoverflow.com/questions/33187086/c-ifstream-object-equals-nullptr-but-it-isnt-a-pointer/33187118

File: CMakeLists.txt (hunk 2 above)
Comment (gruberchris) [reply CMakeLists.txt]: You want to manually replace "main" with your project name each time you start a new project?

File: .gitignore (hunk 1 above)
Comment (gruberchris) [reply .gitignore]: sure I can undo adding in the .gitignore if you don't want one in your template. I am exactly the opposite as you on this opinion but I won't judge ;-)

Comment (ArthurSonzogni) [reply .gitignore]: ;-)

File: CMakeLists.txt (hunk 2 above)
Comment (ArthurSonzogni) [reply CMakeLists.txt]: Yes, I want ;-)

File: src/Shader.cpp (hunk 3 above)
Comment (ArthurSonzogni) [reply src/Shader.cpp]: Yes, this is evaluated, via its `operator bool()`.

See: https://www.cplusplus.com/reference/ios/ios/operator_bool/

File: CMakeLists.txt [hunk 4]
```diff
@@ -14,18 +14,24 @@ add_executable(main
   src/Shader.hpp
   src/Shader.cpp
 )
-set_property(TARGET main PROPERTY CXX_STANDARD 11)
-target_compile_options(main PRIVATE -Wall)
+
+add_custom_command(TARGET ${CMAKE_PROJECT_NAME} POST_BUILD
```
Comment (ArthurSonzogni) [inline CMakeLists.txt]: This is probably going to break if developers are building from the source directory.

I am wondering if we shouldn't instead encode the full path?
I remember this is what I did for the examples of [smk](https://github.com/ArthurSonzogni/smk) here:
https://github.com/ArthurSonzogni/smk/tree/master/examples/assets

Comment (gruberchris) [reply CMakeLists.txt]: I'll try that when I get home to be sure.

Comment (gruberchris) [reply CMakeLists.txt]: Ran this using CLion and seems to have compiled and executed fine 
image

Comment (gruberchris) [reply CMakeLists.txt]: And I did also run  it off the command-line with cmake.exe on Windows and still got the build I was expecting.
image

Comment (gruberchris) [reply CMakeLists.txt]: And, of course, I originally made this with Visual Studio 2022 using CMake and that built and executed fine also.

File: src/Shader.cpp (hunk 3 above)
Comment (gruberchris) [reply src/Shader.cpp]: i just pushed a revert for this concern. Thanks for pointing it out.
